analyzer_test.go

```diff
@@ -86,515 +101,382 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
-							t.Fatalf("Found bad-layer layer dir, it should be removed")
-						}
-						if _, err := ioutil.ReadFile(filepath.Join(layerDir, "metadata.buildpack", "bad-layer.toml")); !os.IsNotExist(err) {
-							t.Fatalf("found bad-layer.toml, it should be removed")
-						}
-						if _, err := ioutil.ReadFile(filepath.Join(layerDir, "metadata.buildpack", "bad-layer.sha")); !os.IsNotExist(err) {
-							t.Fatalf("Found stale bad-layer.sha, it should be removed")
-						}
-					})
+					h.AssertPathDoesNotExist(t, filepath.Join(layerDir, "metadata.buildpack", "launch-cache-not-in-app.toml"))
+					h.AssertPathDoesNotExist(t, filepath.Join(layerDir, "metadata.buildpack", "launch-cache-not-in-app.sha"))
 				})
 
-				when("there are stale cached launch/build layers", func() {
-					it("removes the layer dir and metadata", func() {
-						// copy to layerDir
-						h.RecursiveCopy(t, filepath.Join("testdata", "analyzer", "cached-layers"), layerDir)
+				it("does not restore cache=false layer metadata", func() {
+					_, err := analyzer.Analyze(image, testCache)
+					h.AssertNil(t, err)
 
-						if _, err := analyzer.Analyze(image); err != nil {
-							t.Fatalf("Error: %s\n", err)
-						}
+					h.AssertPathDoesNotExist(t, filepath.Join(layerDir, "metadata.buildpack", "cache-false.toml"))
+					h.AssertPathDoesNotExist(t, filepath.Join(layerDir, "metadata.buildpack", "cache-false.sha"))
+				})
 
-						var err error
-						if _, err = ioutil.ReadDir(filepath.Join(layerDir, "metadata.buildpack", "stale-launch-build")); !os.IsNotExist(err) {
-							t.Fatalf("Found stale stale-launch-build layer dir, it should not exist")
-						}
+				it("restores escaped buildpack layer metadata", func() {
+					_, err := analyzer.Analyze(image, testCache)
+					h.AssertNil(t, err)
 
-						if _, err := ioutil.ReadFile(filepath.Join(layerDir, "metadata.buildpack", "stale-launch-build.toml")); !os.IsNotExist(err) {
-							t.Fatalf("Found stale stale-launch-build.toml, it should be removed")
-						}
+					path := filepath.Join(layerDir, "escaped_buildpack_id", "escaped-bp-layer.toml")
+					got := h.MustReadFile(t, path)
+					want := "[metadata]\n  escaped-bp-layer-key = \"escaped-bp-layer-value\""
 
-						if _, err := ioutil.ReadFile(filepath.Join(layerDir, "metadata.buildpack", "stale-launch-build.sha")); !os.IsNotExist(err) {
-							t.Fatalf("Found stale stale-launch-build.sha, it should be removed")
-						}
-					})
+					h.AssertStringContains(t, string(got), want)
 				})
 
-				when("there cached launch layers that are missing from metadata", func() {
-					it("removes the layer dir and metadata", func() {
-						// copy to layerDir
-						h.RecursiveCopy(t, filepath.Join("testdata", "analyzer", "cached-layers"), layerDir)
-
-						if _, err := analyzer.Analyze(image); err != nil {
-							t.Fatalf("Error: %s\n", err)
-						}
+				it("restores escaped buildpack layer metadata", func() {
```

Comment:
This looks like a copy of the previous test

Reply:
Good catch, removed.